### ISSUE TITLE
ENH: implement Kleene versions of GroupBy any/all kernels for nullable dtypes

### ISSUE THREAD

**Issue by arw2019** [pandas-dev/pandas#37506]
xref   https://github.com/pandas-dev/pandas/issues/37493#issuecomment-718974981

**Comment by phofl** [comment]
Wanted to look into this and tried to understand the implementation for Keene logic in case of ``any/all`` for a Series.  This means, that

```
s = pd.Series([False, pd.NA, False], dtype="boolean")
print(s.any(skipna=False))
```

works and returns ``pd.NA``. But 

```
df= pd.DataFrame({"a": [False, pd.NA, False]}, dtype="boolean")
print(df.any(skipna=False))
```

returns an empty Series. I could not find the reason, why this was not implemted for DataFrames too, when any columns there is from dtype ``boolean``. Should this get implemented for SeriesGroupBy and DataFrameGroupBy?

**Comment by arw2019** [comment]
I haven't gone through the whole path put I think we may want to do this in the cython method

**Comment by phofl** [comment]
For GroupBy that would probably be the best. But we have to fix #37505 before doing this, because this raises for
```
s = pd.Series([False, pd.NA, False], dtype="boolean", index=[0,0, 1])
grouped = s.groupby(level=0)
print(grouped.any(skipna=False))
```
 too. But I meant the regular ``DataFrame.any()`` without groupby.

**Comment by arw2019** [comment]
> For GroupBy that would probably be the best. But we have to fix #37505 before doing this, because this raises for
> 
> ```
> s = pd.Series([False, pd.NA, False], dtype="boolean", index=[0,0, 1])
> grouped = s.groupby(level=0)
> print(grouped.any(skipna=False))
> ```
> 
> too before even reaching the cython part. But I meant the regular `DataFrame.any()` without groupby. This does not work either
> 
> Traceback:
> 
> ```
> Traceback (most recent call last):
>   File "/home/developer/PycharmProjects/pandas/pandas/core/groupby/groupby.py", line 2594, in _get_cythonized_result
>     vals, inferences = pre_processing(vals)
>   File "/home/developer/PycharmProjects/pandas/pandas/core/groupby/groupby.py", line 1368, in objs_to_bool
>     vals = vals.astype(bool)
>   File "/home/developer/PycharmProjects/pandas/pandas/core/arrays/boolean.py", line 388, in astype
>     raise ValueError("cannot convert float NaN to bool")
> ValueError: cannot convert float NaN to bool
> ``

Ah but "boolean" is an Extension array right? I think if we rewrite in terms of a mask that's a fix and it's preferred performance-wise. But in the meantime we can cast to numpy before passing to cython as was done for std in https://github.com/pandas-dev/pandas/pull/37433
